hr_governance/models/governance_circle.py

```diff
@@ -0,0 +1,620 @@
+        """
+        return random.randint(2, 11)
+
+    @api.depends("child_ids")
+    def _compute_is_circle(self):
+        for rec in self:
+            rec.is_circle = len(rec.child_ids) or rec.is_root
+
+    @api.depends(
+        "is_circle",
+        "type_id",
+        "type_id.purpose",
+        "type_id.name",
+        "type_id.expectation",
+        "type_id.authority",
+        "type_id.color",
+    )
+    def _compute_fields_from_type(self):
+        for rec in self:
+            if rec.is_circle:
+                rec.color = rec.color or rec._get_default_color()
+            else:
+                if not rec.type_id:
+                    rec.color = rec.color or 1  # role is white by default
+                    continue
+
+                if not rec.name:
+                    rec.name = rec.type_id.name
+
+                if not rec.color:
+                    rec.color = rec.type_id.color
+
+                if not rec.user_input_purpose:
+                    rec.purpose = rec.type_id.purpose
+
+                if not rec.user_input_expectation:
+                    rec.expectation = rec.type_id.expectation
+
+                if not rec.user_input_authority:
+                    rec.authority = rec.type_id.authority
+
+    @api.depends("type_id.type")
+    def _compute_is_editable(self):
+        for rec in self:
+            rec.is_editable = not rec.type_id or rec.type_id.type == "template"
+
+    @api.depends("is_circle")
+    def _compute_is_addable(self):
+        for rec in self:
+            is_single_mode_on = str2bool(
+                self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("hr_governance.governance_single_assignee_mode")
+            )
+            if is_single_mode_on:
+                rec.is_addable = not rec.is_circle and not rec.member_ids
+            else:
+                rec.is_addable = True
+
+    @api.depends("member_ids")
+    def _compute_member_count(self):
+        for rec in self:
+            rec.member_count = len(rec.member_ids)
+
+    @api.constrains("parent_id")
+    def _check_parent_set(self):
+        for rec in self:
+            if not rec.is_root and not rec.parent_id:
+                raise UserError(_("Parent must be set"))
+
+    @api.model_create_multi
+    def create(self, vals):
+        for val in vals:
+            context = self.env.context
+            # auto create structuring roles for circle
+            if context.get("default_is_circle", False):
+                circle = self.new(val)
+                new_roles_vals = circle.type_id._get_structuring_role_vals()
+                existing_child_ids = val.get("child_ids", [])
+                new_child_ids = existing_child_ids + [
+                    (0, 0, val) for val in new_roles_vals
+                ]
+                val["child_ids"] = new_child_ids
+
+        records = super().create(vals)
+
+        return records
+
+    def _get_fields(self):
```

Comment:
I would rename this method to make it easier to understand what it does. For example: what are these fields used for?